pytest_reportportal/listener.py

```diff
@@ -53,17 +47,15 @@ def pytest_runtest_protocol(self, item):
         """
         self._add_issue_id_marks(item)
         item_id = self.py_test_service.start_pytest_item(item)
-        if PYTEST_HAS_LOGGING_PLUGIN:
-            # This check can go away once we support pytest >= 3.3
-            with patching_logger_class():
-                with _pytest.logging.catching_logs(self._log_handler,
-                                                   level=self._log_level):
-                    yield
-        else:
-            yield
+        with patching_logger_class():
+            with _pytest.logging.catching_logs(self._log_handler,
+                                               level=self._log_level):
+                yield
         # Finishing item in RP
         self.py_test_service.finish_pytest_item(
             item, item_id, self.result or 'SKIPPED', self.issue or None)
+        # Flush log buffer
+        self.py_test_service.rp.terminate()
```

Comment:
This line should go into the pytest_runtest_teardown hook as described in the issue.